pkg/registry/bundlegraphloader_test.go

```diff
@@ -312,3 +313,71 @@ func TestBundleGraphLoader(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSkipPatchCandidate(t *testing.T) {
+	tests := []struct{
+		name     string
+		added     string
+		compare  string
+		expected bool
+	}{
+		{
+			name: "equal versions",
+			added: "0.0.0",
+			compare: "0.0.0",
+			expected: false,
+		},
+		{
+			name: "do not accept different major/minor version",
+			added: "0.0.0",
+			compare: "0.0.0",
+			expected: false,
+		},
+		{
+			name: "do not accept smaller patch version",
+			added: "0.0.0",
+			compare: "0.0.1",
+			expected: false,
+		},
+		{
+			name: "accept larger patch version",
+			added: "0.0.1",
+			compare: "0.0.0",
+			expected: true,
+		},
+		{
+			name: "accept patch version without pre-release",
```

Comment:
@ankitathomas if it is the following data,
```console
{
			name: "do not accept patch version with pre-release",
			added: "0.0.0-1",
			compare: "0.0.0",
			expected: false,
		},
```
is it correct? I think so. thanks

Reply:
@ankitathomas To add on to this, if `f(a, c) -> !f(c, a)` and `!f(a, c) -> f(c, a)`, then this test could automatically verify both for each case.

Reply:
I've added checking for the absence of a commutative edge